File: Samples/VB.Net/VBDotNet/ICustomerDlg.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports SalesMatePlusLib

Namespace CSharpSDKTest
	Partial Public Class ICustomerDlg
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Cstmrdtlsbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles Cstmrdtlsbtn.Click
			Dim m_Customer As ICustomer

			m_Customer = New CustomerClass()

			CustIdlbl.Visible = False

			FldNmelbl.Visible = False

			Dim strCusId, strFldNme As String

			strCusId = custIdTxt.Text

			strFldNme = FldNmeTxt.Text

			If (strCusId = "") AndAlso (strFldNme = "") Then
				CustIdlbl.Visible = True
				FldNmelbl.Visible = True

			ElseIf strCusId = "" Then
				CustIdlbl.Visible = True

			ElseIf strFldNme = "" Then
				FldNmelbl.Visible = True

			Else
				Dim VarVal As Object = 0

				m_Customer.GetCustomerDetails(strCusId, strFldNme, VarVal)

				Dim strMsg As String = VarVal.ToString()

				MessageBox.Show(strMsg)
			End If
		End Sub

		Private Sub FindCustomerbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles FindCustomerbtn.Click
			Dim m_Customer As ICustomer

			m_Customer = New CustomerClass()

			Dim strCusId As String=""

			m_Customer.InvokeFindCustomerDlg(strCusId)
		End Sub

		Private Sub clsbtn_Click(ByVal sender As Object, ByVal e As EventArgs) Handles clsbtn.Click
			Me.Close()
		End Sub
	End Class
End Namespace

File: Samples/C#.Net/CSharpSDKTest/ICustomerDlg.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using SalesMatePlusLib;

namespace CSharpSDKTest
{
    public partial class ICustomerDlg : Form
    {
        public ICustomerDlg()
        {
            InitializeComponent();
        }

        private void Cstmrdtlsbtn_Click(object sender, EventArgs e)
        {
            ICustomer m_Customer;

            m_Customer = new CustomerClass();

            CustIdlbl.Visible = false;

            FldNmelbl.Visible = false;

            string strCusId, strFldNme;

            strCusId = custIdTxt.Text;

            strFldNme = FldNmeTxt.Text;

            if ((strCusId == "") && (strFldNme == ""))
            {
                CustIdlbl.Visible = true;
                FldNmelbl.Visible = true;
            }

            else if (strCusId == "") 
            {
                CustIdlbl.Visible = true;
            }

            else if (strFldNme == "")
            {
                FldNmelbl.Visible = true;
            }

            else
            {
                object VarVal = 0;

                m_Customer.GetCustomerDetails(strCusId, strFldNme, ref VarVal);

                string strMsg = VarVal.ToString();

                MessageBox.Show(strMsg);
            }
        }

        private void FindCustomerbtn_Click(object sender, EventArgs e)
        {
            ICustomer m_Customer;

            m_Customer = new CustomerClass();

            string strCusId="";

            m_Customer.InvokeFindCustomerDlg(ref strCusId);
        }

        private void clsbtn_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}